charts/castai-agent/templates/secret.yaml

```diff
@@ -7,7 +8,8 @@ metadata:
   labels:
     {{- include "castai-agent.labels" . | nindent 4 }}
 data:
-  API_KEY: {{ required "apiKey must be provided" .Values.apiKey | b64enc | quote }}
+  API_KEY: {{ .Values.apiKey | b64enc | quote }}
```

Comment:
`required` check is no longer performed. If a user does not `--set=castai.apiKey=$CASTAI_API_TOKEN`, then it's expected the user have their own secret and it is being accessed through `additionalEnv` field.

Reply:
Yeah looks like we need either the secret specified directly or a secret ref. However, leave the flexible `additionalEnv` :+1:

Reply:
Something like that in `values.yaml`, which could also be validated in helm helpers that either one of those exist:
```
apiKey:
  # Either value or secretKeyRef must be specified.
  #
  # value: "my hardcoded secret"
  # secretKeyRef:
  #  name: my-custom-secret
  #  key: FOOBAR
```